### PR TITLE
[C-2680] Attribution Modal

### PR DESCRIPTION
**Pull request by amendelsohn**
### Description

Screenshot 2023-07-20 at 7 13 51 PM

### Feature Flags ###

`UPLOAD_REDESIGN_ENABLED`

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/amendel-attribution-modal-form

File: packages/web/src/pages/upload-page/forms/AttributionModalForm.module.css [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+.license {
+  display: flex;
+  flex-direction: column;
+  gap: var(--unit-2);
+  padding: var(--unit-4);
+  border-radius: 8px;
+  border: 1px solid var(--neutral-light-8);
+  background: var(--neutral-light-10);
+  text-transform: capitalize;
```
Comment (sliptype) [inline packages/web/src/pages/upload-page/forms/AttributionModalForm.module.css:52]: The designs don't have this text-transform from what I can tell

File: packages/web/src/pages/upload-page/forms/AttributionModalForm.module.css [hunk 2]
```diff
@@ -0,0 +1,57 @@
+.row {
```
Comment (sliptype) [inline packages/web/src/pages/upload-page/forms/AttributionModalForm.module.css:1]: I'm a fan of this but we could generalize it even further by putting these into their own file, like common classes

File: packages/web/src/pages/upload-page/forms/AttributionModalForm.tsx [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+            })}
+          >
+            {messages.derivativeWorks.header}
+          </div>
+          <SegmentedControl
+            fullWidth
+            defaultSelected={derivativeWorks}
+            // @ts-ignore boolean support works
+            options={derivativeWorksValues}
+            // @ts-ignore
```
Comment (sliptype) [inline packages/web/src/pages/upload-page/forms/AttributionModalForm.tsx:294]: Pretty annoying that we have to do this

**Review by sliptype — APPROVED** [review]
Looks great! Super clean component structure and styles

File: packages/web/src/pages/upload-page/forms/AttributionModalForm.tsx (hunk 3 above)
Comment (amendelsohn) [reply packages/web/src/pages/upload-page/forms/AttributionModalForm.tsx:304]: Yeah it's fixable but involves rewriting a bunch of stuff I think. I tried fooling with the types, but we'd have to ts-ignore in the component instead. Maybe better? idk

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/amendel-attribution-modal-form

**Pull request merged**